Purchases/Logging/Logger.swift

```diff
@@ -37,32 +37,38 @@ class Logger {
 
     private static let frameworkDescription = "Purchases"
 
-    static func log(level: LogLevel, message: String) {
-        guard self.logLevel.rawValue <= level.rawValue else { return }
-        logHandler(level, message)
-    }
-
-    static func log(level: LogLevel, intent: LogIntent, message: String) {
-        let messageWithPrefix = "\(intent.suffix) \(message)"
-        Logger.log(level: level, message: messageWithPrefix)
-    }
-
     static func debug(_ message: String) {
         log(level: .debug, intent: .info, message: message)
     }
 
+    static func debug(_ message: CustomStringConvertible) {
+        log(level: .debug, intent: .info, message: message.description)
+    }
+
```

Comment:
I added a duplicate method for each of them to help migrate between approaches. However, once all strings are moved to the new format, we can remove the `String` versions of these methods